### PR TITLE
improve type definition

### PR DESCRIPTION
**Pull request by troy351**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] commit message follows commit guidelines

##### Description of change

- Add missing type definition
- Add additional comment for type definition
- Fixed demos page `execute` button will always scroll the editor to top which has bad experience
- Fixed tooltip should be default `display: none;` when initialized

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -792,32 +1213,84 @@ declare namespace G2 {
...
+    rowValue: ChartValue;
+    /**
+     * 分面总行数
+     */
+    rows: number;
+  }
+
+  class Chart extends BaseView {
+    constructor(option: ChartProps);
     legend(option: boolean): this;
```
Comment (hsuanxyz) [inline src/index.d.ts]: `interact(..)` 也漏掉了

Comment (troy351) [reply src/index.d.ts]: 文档里没写。。

File: src/index.d.ts [hunk 2]
```diff
@@ -410,46 +418,220 @@ declare namespace G2 {
       | 'bottom-left'
```
Comment (hsuanxyz) [inline src/index.d.ts]: `top-bottom` 写错了吧，应该是 `top-right`

File: src/index.d.ts (hunk 1 above)
Comment (hsuanxyz) [reply src/index.d.ts]: 这个可以这么来

```ts
export enum InteractionsType {
  Brush = 'brush',
  Drag = 'drag',
  Zoom = 'zoom',
  ScrollBar = 'scroll-bar',
  ShapeSelect = 'shape-select',
  Slider = 'slider',
}

export interface Interactions {
  [InteractionsType.Brush]: BrushConfig;
  [InteractionsType.Drag]: DragConfig;
  [InteractionsType.Zoom]: ZoomConfig;
  [InteractionsType.ScrollBar]: ScrollBarConfig;
  [InteractionsType.ShapeSelect]: ShapeSelectConfig;
  [InteractionsType.Slider]: SliderConfig;
}

class Chart extends BaseView {
  interact<T extends InteractionsType>(type: InteractionsType, config?: Interactions[T])
}
```

File: src/index.d.ts (hunk 2 above)
Comment (troy351) [reply src/index.d.ts]: 其实下面的`vertica`也打错了

Comment (hsuanxyz) [reply src/index.d.ts]: 嗯，手写的确无法避免这类问题，等用 ts 重构完才能完全解决

**Comment by dengfuping** [comment]
- @troy351 Sorry to close it, because we are working on version `4.x` with `TypeScript` support.

**Comment by troy351** [comment]
@dengfuping Any `4.x` roadmap?

**Comment by dengfuping** [comment]
- @troy351 Chinese version is available only in [here](https://www.yuque.com/antv/g2-docs/nmzh8r). We need to provide more detailed roadmap with Chinese and English version.

**Pull request closed without merging**